### DISCUSSION TITLE
Is it not possible to use special characters for field keys?

### DISCUSSION THREAD

**Discussion by zlzonchan** [GodotNuts/GodotFirebase#446] [Q&A] [answered]
I added a feature that allows users to create their own puzzles and to upload them to Firestore. The field key is set as the title string specified by the user, but when I tried to update fields with keys containing '?' or '....', it resulted in an error. However, there was no issue when adding these directly through the Firebase console. Why does this happen?

**Comment by WolfgangSenff** [comment, marked as answer]
Not completely sure, but it's possible either they updated the rules on what's allowed as characters of keys and we didn't, or perhaps the REST API just won't allow them, not sure. That said, why would you allow them to set the _key_ for the values? Don't you want to set the key to "title" and the value to whatever they have written, so you can actually reference it in your game? It sounds like you have something backwards perhaps, unless I'm missing why you'd want to allow someone to potentially break your database by injecting code or other things into it in that fashion.

**Reply by zlzonchan** [reply]
I agree. This is my first time managing database so I think I missed that point. I'll implement these changes. Thanks!
